VersionControlUI/src/commands/CHistory.cpp

```diff
@@ -41,11 +41,13 @@ CHistory::CHistory() : CommandWithFlags{"history", {{"accumulate"}}, true, false
 {
 }
 
-Interaction::CommandResult* CHistory::executeNamed(Visualization::Item* /*source*/, Visualization::Item* target,
+Interaction::CommandResult* CHistory::executeNamed(Visualization::Item* source, Visualization::Item*,
 												  const std::unique_ptr<Visualization::Cursor>& /*cursor*/,
 												  const QString& name, const QStringList& attributes)
 {
-	Model::TreeManager* headManager = target->node()->manager();
+	auto ancestorWithNode = source->findAncestorWithNode();
```

Comment:
This could return `nullptr`, in which case you should make sure not to do anything. Ideally, this is caught already in the `canExecute` method.